packages/kbn-alerts-ui-shared/src/alerts_search_bar/index.tsx

```diff
@@ -119,7 +119,11 @@ export const AlertsSearchBar = ({
     displayStyle: 'inPage',
     showFilterBar,
     onQuerySubmit: onSearchQuerySubmit,
-    onFiltersUpdated,
+    onFiltersUpdated: (newFilters) => {
+      // filterManager.setFilters populates filter.meta so filter pill has pretty title
+      dataService.query.filterManager.setFilters(newFilters);
```

Comment:
This seems to be an implementation detail in the unified search bar (although I don't really understand how this filterManger works so maybe I'm wrong). Therefore, I would prefer to have some kind of prop in the unified search bar component that executes this internally onFiltersUpdate but I understand that we are overwriting the default behaviour completely, is that right?

Reply:
> I understand that we are overwriting the default behaviour completely, is that right?

Correct.

There is mechanism in [unified search bar](https://github.com/elastic/kibana/blob/main/src/plugins/unified_search/public/search_bar/create_search_bar.tsx) to set filters using filterManager when there is no `onFiltersUpdated` method. 
```typescript
// Respond to user changing the filters
const defaultFiltersUpdated = (
  queryService: QueryStart,
  onFiltersUpdated?: (filters: Filter[]) => void
) => {
  return (filters: Filter[]) => {
    if (onFiltersUpdated) {
      onFiltersUpdated(filters);
    } else {
      queryService.filterManager.setFilters(filters);
    }
  };
};
``` 

However I tried removing `onFiltersUpdated` from our plugin to use above method but it didn't render the filter. 🤷‍♀️

Reply:
My ideas would be to update that fn to be something like:

```typescript
// Respond to user changing the filters
const defaultFiltersUpdated = (
  queryService: QueryStart,
  onFiltersUpdated?: (filters: Filter[]) => void
  forceExecuteDefaultFiltersUpdate?: boolean // naming is hard, but you get the idea
) => {
  return (filters: Filter[]) => {
    if (!onFiltersUpdated || forceExecuteDefaultFiltersUpdate)
      queryService.filterManager.setFilters(filters);
    }

    if (onFiltersUpdated) {
      onFiltersUpdated(filters);
    }
  };
};
``` 
`forceExecuteDefaultFiltersUpdate` would be true/false depending on the value of a react prop so we can set it to true when wrapping the search bar on our side. I hope it makes sense. This would avoid having to wrap the onFiltersUpdate fn with the magic `filterManager.setFilters` call. But if it's being overwritten I guess we might not have other option

Reply:
[Fixed in unified search plugin ](https://github.com/elastic/kibana/pull/193623/commits/eb0ebd84e4d4631de75674d5df82533d7929e5e8)